extensions/gdx-tools/src/com/badlogic/gdx/tools/particleeditor/EffectPanel.java

```diff
@@ -165,7 +166,7 @@ void openEffect (boolean mergeIntoCurrent) {
 			JOptionPane.showMessageDialog(editor, "Error opening effect.");
 			return;
 		}
-		for (ParticleEmitter emitter : editor.effect.getEmitters()) {
+		for (ParticleEmitter emitter : new Array.ArrayIterator<>(editor.effect.getEmitters())) {
```

Comment:
This looks like a major improvement if there's concurrent modification happening.

Reply:
I got nested iterators error while trying to read some particle effects with multiple emitters.

Reply:
Then this seems like the thing to use to fix it!